### ISSUE TITLE
Style/RedundantReturn need white list

### ISSUE THREAD

**Issue by QWYNG** [rubocop-hq/rubocop#7587]
This issue is updated version from #7539 (with a little correction around my misunderstandings) 

With Rails controller, we sometimes see codes as below:

```ruby
  def show
    if cond?
      redirect_to posts_url
      return
    end
  end
```
Here, this `return` is not needed generally speaking, but some developers would prefer  
 to put it as it can avoid potential DoubleRenderErrors 
For example, another developer might modify as below.

```ruby
def show
  if cond?
    redirect_to posts_url
    return
  end

  # ========== snip =========

  render :some_view
end
```
Good point about this code is that we can make sure to avoid feature DoubleRenderError.
Now, Rubocop this `return` violate by Style/RedundantReturn.

I'd suggest a solution
add AllowedMethods option
```
  AllowedMethods:
    - redirect_to
```

In most cases, redirect_to and render need this option

**Comment by bbatsov** [comment]
So, what exactly is the whitelist going to act on? The preceding method call? This seems really weird to me. If it's double render errors that we want to avoid won't it be better to simply write a cop that checks for such problems?

**Comment by QWYNG** [comment]
@bbatsov 
> So, what exactly is the whitelist going to act on? The preceding method call? This seems really weird to me. If it's double render errors that we want to avoid won't it be better to simply write a cop that checks for such problems?

like this?
```
def show
  if cond?
    redirect_to posts_url
    return
  end

  # ========== snip =========

  render :some_view
  ^^^^^^^^^^^^^^^^^^^DoubleRender Error detected.
  
end
```

Certainly, This cop should place in rubocop-rails.

**Comment by bbatsov** [comment]
Yep.
